src/main/java/org/jabref/gui/search/GlobalSearchResultDialog.java

```diff
@@ -53,6 +57,10 @@ public GlobalSearchResultDialog(UndoManager undoManager) {
     private void initialize() {
         viewModel = new GlobalSearchResultDialogViewModel(preferencesService);
 
+        SearchResultDialogSearchBar searchBar = new SearchResultDialogSearchBar(libraryTabContainer, stateManager, preferencesService, undoManager);
```

Comment:
I like the idea of having this as a separate component,  but inside you have bascially duplicate functionality/content to the normal "SearchBar". Can the normal search bar also be adapted to use this new component? 
DRY (Don't Repeat Yourself) Principle

Reply:
I tried to avoid these duplications, but I don't know how to handle them. Should I create a new class containing some static methods and call them? Or what do you suggest? The only difference between the two classes is the removal of two buttons (`keepSearchString`, `openGlobalSearchButton`) and a change in the listener to listen to the newly introduced property `activeGlobalSearchQuery`.

Reply:
Pass the listener as Consumer parameter  and maybe pass an enum e.g.  NORMAL_SEARCH, GLOBAL_SEARCH and adapt the buttons accordingly